php/async/Exchange.php

```diff
@@ -157,6 +157,12 @@ public function fetch($url, $method = 'GET', $headers = null, $body = null) {
         }) ();
     }
 
+    public function fetch_currencies($params = array()) {
+        return React\Async\async(function () use ($params) {
+            return parent::fetch_currencies($params);
```

Comment:
@ttodua why call the parent instead of simply returning `this->$currencies` ? I don't think we need to overcomplicate here

Reply:
because we have other logic in upper, which can be changed or etc, so this method shouldnt matter what we do in parent, just return parent's fC result. why it would complicate?

Reply:
either, returning `$this->currencies` is not correct as it might be undefined or whatever idk, that's why we have other logic in parent `return $this->currencies ? $this->currencies : array();`

Reply:
> either, returning `$this->currencies` is not correct as it might be undefined or whatever idk, that's why we have other logic in parent `return $this->currencies ? $this->currencies : array();`

yeah that is why I mean, but fine let's keep it then